File: Kth Smallest Element in a BST/sol2.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int ans = 0;
    int process_count = 0;
    void inorder(TreeNode* root, int k){
        int temp;
        if(root ==NULL){
            return;
        }
        inorder(root->left, k);
        temp = root->val;
        process_count++;
        if(process_count==k){
            ans=temp;
            return;
        }
        inorder(root->right, k);
    }
    int kthSmallest(TreeNode* root, int k) {
        inorder(root,k);
        return ans;
    }
};

File: Maximum Depth of Binary Tree/sol2.py

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    max_height = 0
    curr = 0
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if(root==None):
            return self.max_height
        self.curr+=1
        if(self.curr>self.max_height):
            self.max_height = self.curr
        self.maxDepth(root.left)
        self.maxDepth(root.right)
        self.curr-=1
        
        return self.max_height
